### DISCUSSION TITLE
tutorials/yolov8-increase-accuracy/

### DISCUSSION THREAD

**Discussion by giscus** [Y-T-G/Yasins-Keep#6] [Giscus]
# tutorials/yolov8-increase-accuracy/

Get over 10% more mAP in small object detection by exploiting YOLOv8 pose models while training.

https://y-t-g.github.io/tutorials/yolov8-increase-accuracy/

**Comment by Jaxys10** [comment]
Thank you for the tutorial! I learned a lot from it. I do the same as your way with my own dataset(85% are small targets).In the first 10 epochs, the map50 did indeed improve significantly.But the final effect is not as good as the original model.Is this related to the dataset? My dataset is grayscale images.

**Reply by Y-T-G** [reply]
How many epochs did you run it for?

**Reply by Y-T-G** [reply]
You should run it using the same configuration as you did originally. If you used a `yolov8m`, you should use `yolov8m-pose`. All the training configuration should be the same. The only thing you need to change is add `box=0.01` and`dfl=0.01`.

**Reply by Jaxys10** [reply]
Thank you for your reply.The epoch is set to 100. I used a yolov8s as the original model and got map50 of 0.674.When I use yolov8s-pose as the model and got map50 of 0.64.Yolov8s-pose training stopped in 72 epochs because I set the patience to 30. I did change the box=0.01 and dfl=0.01.

**Reply by Jaxys10** [reply]
Is it related to the settings of imgsz? To achieve better performance, I set imgsize to 1280.All the training configuration  are the same except for box and dfl.

**Reply by Y-T-G** [reply]
You can try using `box=0.1` and `dfl=0.1`

**Reply by Y-T-G** [reply]
Is `0.64` the highest mAP50 you got in results.csv? The default behavior in ultralytics is that it will prioritize mAP50-95 when saving the `best.pt` so it might ignore a better mAP50 checkpoint.

**Reply by Y-T-G** [reply]
Also, this will affect the `patience` too because it judges improvements based on `mAP50-95` score. Better to disable patience.

**Reply by Jaxys10** [reply]
Yeah,0.64 is the highest mAP50 in my results.csv.I changed model fitness weights from [P, R, mAP@0.5, mAP@0.5:0.95] to [0.0, 0.0, 0.99, 0.01]. I tried using box=0.1 and dfl=0.1.The mAP50 increased to 0.648,but the results still lower than original model.

**Reply by Y-T-G** [reply]
What if you don't apply any weights to box and dfl?

**Reply by Jaxys10** [reply]
The best mAP50 become 0.623.

**Reply by Y-T-G** [reply]
Do you have background images? Did the conversion to keypoint skip those images?

**Reply by Jaxys10** [reply]
Yes, there are a few background images.From train_batch, it seems that there is no problem with the labels.

**Reply by Y-T-G** [reply]
You could try smaller imgsz and see.

Or you could try using your old model's (the one with 67 mAP) checkpoint:
```python
model = YOLO("yolov8s-pose.pt").load("old_model.pt")
model.train(..., box=0.01, dfl=0.01, optimizer="SGD", lr=0.001)
```

**Reply by Jaxys10** [reply]
Thank you for your suggestion. I will try it.

**Comment by tomer196** [comment]
Thanks for this useful guide, I will surely try it out soon.
What is the effect on large object detection in a mixed dataset of small and large objects?

**Reply by Y-T-G** [reply]
You could try it. It might not be much better than the normal model.

**Comment by LingjieBao1998** [comment]
I have a question？ Are you trying 5 key points of detection, which include the center point, top right, bottom right, left down, and left up.

**Reply by Y-T-G** [reply]
I didn't try. The four corners would already be found by the bounding box, so it's not necessary. You can just use 1 keypoint which is the center.

**Reply by LingjieBao1998** [reply]
Thanks for your reply! the mode of `yolo pos` is bbox detection first then keypoints detection, i‘m wonder about the extra four corners help detection.

**Reply by Y-T-G** [reply]
You can try

**Reply by LingjieBao1998** [reply]
Thank you for your response! Are there alternative approaches for small object detection? While I experimented with SAHI, I found that its slicing mechanism often leads to truncated objects​.

**Reply by Y-T-G** [reply]
You can train P2 model

https://github.com/Y-T-G/community/blob/main/cfg/detect/yolo11-p2.yaml
